classes/stock/StockAvailable.php

```diff
@@ -698,6 +740,33 @@ public static function outOfStock($id_product, $id_shop = null)
         return (int) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue($query);
     }
 
+    /**
+     * @param int $id_product
+     * @param int id_product_attribute Optional
+     * @param int $id_shop Optional
+     *
+     * @return bool|string
+     */
+    public static function getLocation($id_product, $id_product_attribute = null, $id_shop = null)
+    {
+        if (false === Validate::isUnsignedId($id_product)) {
+            return false;
```

Comment:
Same here, I think we should throw an exception instead :) it can be a simple InvalidArgumentException but avoid missing multiple return types to manage wrong inputs. Also, you can validate other fields.